src/axom/mir/EquiZAlgorithm.hpp

```diff
@@ -0,0 +1,1398 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for internals.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+#ifndef AXOM_MIR_EQUIZ_ALGORITHM_HPP_
+#define AXOM_MIR_EQUIZ_ALGORITHM_HPP_
+
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/mir.hpp"
+#include "axom/slic.hpp"
+
+// Include these directly for now.
+#include "axom/mir/MIRAlgorithm.hpp"
+#include "axom/mir/utilities/ExtractZones.hpp"
+#include "axom/mir/utilities/MergeMeshes.hpp"
+#include "axom/mir/utilities/NodeToZoneRelationBuilder.hpp"
+#include "axom/mir/utilities/RecenterField.hpp"
+#include "axom/mir/utilities/ZoneListBuilder.hpp"
+#include "axom/mir/views/dispatch_coordset.hpp"
+#include "axom/mir/views/MaterialView.hpp"
+
+#include <conduit/conduit.hpp>
+
+// RAJA
+#if defined(AXOM_USE_RAJA)
+  #include "RAJA/RAJA.hpp"
+#endif
+
+#include <algorithm>
+#include <string>
+
+// This macro makes the EquiZ algorithm split processing into clean/mixed stages.
+#define AXOM_EQUIZ_SPLIT_PROCESSING
+
+// Uncomment to save inputs and outputs.
+// #define AXOM_EQUIZ_DEBUG
+
+#if defined(AXOM_EQUIZ_DEBUG)
+  #include <conduit/conduit_relay_io_blueprint.hpp>
+#endif
+
+namespace axom
+{
+namespace mir
+{
+using MaterialID = int;
+using MaterialIDArray = axom::Array<MaterialID>;
+using MaterialIDView = axom::ArrayView<MaterialID>;
+using MaterialVF = float;
```

Comment:
Is `float` a design choice over `double`?

Reply:
Yes, but I think I could change it to use the material view's FloatType for MaterialVF so it stays true to the types specified in the material view. I think I had some compiler problems in this code but I could try again.